src/Core/Framework/Adapter/Twig/TemplateFinder.php

```diff
@@ -126,23 +126,32 @@ private function getSourceBundleName(string $source): ?string
     }
 
     /**
-     * @return string[]
+     * Gets the final namespace hierarchy for template resolution
+     *
+     * Transforms priority-based ordering to a list of namespace names.
+     * Priority values are discarded after serving their sorting purpose.
+     *
+     * @return string[] Ordered namespace names (last element = highest priority)
```

Comment:
```suggestion
     * @return list<string> Ordered namespace names (last element = highest priority)
```

the generic type annotation is more accurate. `string[]` can also be `array<string, string>` or `array<int, string>`

I don't know if `list` matches. but we should avoid the `[]` annotation

Reply:
What telgi says ⬆️